### PR TITLE
Fixes #3266 Report lines changed

### LINKED ISSUE #3266: Show Number of Lines Yanked/Pasted
In Vim when you yank or paste more than 1 line you get an echo in the status line (?) of the number of lines yanked/pasted.

image

I'd like to see this same functionality so I can easily indent my code after pasting. I use this to paste my visually selected block, then I'd like to be able to `X>>` to indent the X pasted lines.

### PR DESCRIPTION
**Pull request by xconverge**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,11 @@
 <p align="center"><strong>Vim emulation for Visual Studio Code</strong></p>
 
 [![http://aka.ms/vscodevim](https://vsmarketplacebadge.apphb.com/version/vscodevim.vim.svg)](http://aka.ms/vscodevim)
+
 [![](https://vsmarketplacebadge.apphb.com/installs-short/vscodevim.vim.svg)](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim)
+
 [![https://travis-ci.org/VSCodeVim/Vim](https://travis-ci.org/VSCodeVim/Vim.svg?branch=master)](https://travis-ci.org/VSCodeVim/Vim)
+
```
Comment (jpoon) [inline README.md]: Prettier goes crazy on the README. Can you undo these line breaks?

File: README.md [hunk 2]
```diff
@@ -358,6 +361,7 @@ Configuration settings that have been copied from vim. Vim settings are loaded i
...
 | vim.timeout      | Timeout in milliseconds for remapped commands                                                                                                                                                                                                                         | Number  | 1000          |
 | vim.whichwrap    | Controls wrapping at beginning and end of line. Comma-separated set of keys that should wrap to next/previous line. Arrow keys are represented by `[` and `]` in insert mode, `<` and `>` in normal and visual mode. To wrap "everything", set this to `h,l,<,>,[,]`. | String  | ``            |
+| vim.report       | Threshold for reporting number of lines changed.                                                                                                                                                                                                                      | Number  | 2             |
```
Comment (jpoon) [inline README.md]: maybe `vim.statusline.reportLinesChangedThreshold`?

**Comment by jpoon** [comment]
Closes #3266

File: README.md (hunk 2 above)
Comment (xconverge) [reply README.md]: This is a direct translation from vim though http://vimdoc.sourceforge.net/htmldoc/options.html#'report'

File: src/statusBar.ts [hunk 3]
```diff
@@ -51,6 +51,28 @@ class StatusBarImpl implements vscode.Disposable {
     this._wasRecordingMacro = isRecordingMacro;
   }
 
+  public ReportLinesChanged(numLinesChanged: number, mode: ModeName) {
```
Comment (jpoon) [inline src/statusBar.ts]: Can you move these to another class/file? I'd like `statusBar` without having to maintain any state or logic of what the status bar should say.

Comment (jpoon) [reply src/statusBar.ts]: Not sure where the best spot of this would be though.

File: README.md (hunk 2 above)
Comment (jpoon) [reply README.md]: Oh my, I did not know this existed. Ok, nevermind :)

File: src/statusBar.ts (hunk 3 above)
Comment (xconverge) [reply src/statusBar.ts]: I poked around neovim to get some inspiration, misc.1.c...so any opposition for me throwing it in util.ts?

Comment (xconverge) [reply src/statusBar.ts]: I also dont need the functions, they were just incase I wanted to clean up the prints a little/handle the non plural cases

**Pull request merged**